resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/spark-base/entrypoint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Check whether there is a passwd entry for the container UID
+myuid=$(id -u)
+mygid=$(id -g)
+uidentry=$(getent passwd $myuid)
+
+# If there is no passwd entry for the container UID, attempt to create one
+if [ -z "$uidentry" ] ; then
+    if [ -w /etc/passwd ] ; then
+        echo "$myuid:x:$myuid:$mygid:anonymous uid:$SPARK_HOME:/bin/false" >> /etc/passwd
```

Comment:
The problem arises because OpenShift runs its containers with an anonymous UID that has no passwd entry.  Spark will crash if it can't find a passwd entry.